gradle-plugin/src/main/groovy/org/groovymc/modsdotgroovy/gradle/MDGExtension.groovy

```diff
@@ -2,135 +2,141 @@ package org.groovymc.modsdotgroovy.gradle
 
 import groovy.transform.CompileStatic
 import groovy.transform.PackageScope
+import org.gradle.api.Action
 import org.gradle.api.Project
+import org.gradle.api.artifacts.ConfigurablePublishArtifact
+import org.gradle.api.file.FileCollection
 import org.gradle.api.model.ObjectFactory
-import org.gradle.api.provider.MapProperty
+import org.gradle.api.provider.ListProperty
 import org.gradle.api.provider.Property
 import org.gradle.api.provider.SetProperty
+import org.gradle.api.tasks.SourceSet
 import org.groovymc.modsdotgroovy.core.Platform
 
 import javax.inject.Inject
 
 @CompileStatic
-class MDGExtension {
-    @PackageScope static final String NAME = 'modsDotGroovy'
-
-    private final MapProperty<Platform, String[]> platforms
-    private final SetProperty<String> environmentBlacklist
+abstract class MDGExtension {
     private final Property<Boolean> setupDsl
     private final Property<Boolean> setupPlugins
     private final Property<Boolean> setupTasks
-    //private final transient Project project
+    private final ListProperty<Platform> platforms
+    private final MDGConversionOptions conversionOptions
+    private final Property<FileCollection> modsDotGroovyFile
 
-    @Inject
-    MDGExtension(ObjectFactory objects, Project project) {
-        this.platforms = objects.mapProperty(Platform, String[])
-        this.platforms.convention(inferPlatforms(project))
+    Property<Boolean> getSetupDsl() {
```

Comment:
These were previously explicit getters and setters for IDE support in old IntelliJ. It looks like there's no setters anymore as IntelliJ's good now, which means this could be further simplified to final Groovy properties instead (e.g.: `final Property<Boolean> setupDsl`)